src/main/java/ml/shifu/shifu/core/dtrain/wnd/EmbedLayer.java

```diff
@@ -120,7 +125,19 @@ public void write(DataOutput out) throws IOException {
      */
     @Override
     public void readFields(DataInput in) throws IOException {
-        // TODO Auto-generated method stub
+        int embedLayerSize = in.readInt();
+        if(this.embedLayers == null) {
+            this.embedLayers = new ArrayList<EmbedFieldLayer>();
+        }
+        for(int i = 0; i < embedLayerSize; i++) {
+            if(this.embedLayers.get(i) == null) {
+                this.embedLayers.add(i, new EmbedFieldLayer());
+            }
+            this.embedLayers.get(i).readFields(in, this.serializationType);
+        }
 
+        while(this.embedLayers.size() > embedLayerSize) {
+            this.embedLayers.remove(embedLayerSize);
+        }
```

Comment:
Why need remove in serialization?

Reply:
This is just to make sure de-serialized list has same size as expected. There will not be any remove action if this layer if initialized properly.